can_tools/scrapers/base.py

```diff
@@ -383,37 +431,73 @@ def _normalize(self):
         df = self.normalize(data)
         return self._store_clean(df)
 
+    def _validate_time_series(self, df) -> [List[Exception]]:
+        """Do checks only applicable to time series data"""
+        issues = []
+        if (df["measurement"] == "cumulative").sum() > 0:
+            bad = timeseries.values_increasing_over_time(df)
+            for variable, dates in bad.items():
+                err = ValidateDecreaseInCumulativeVariableError(variable, dates)
+                issues.append(err)
+
+        return issues
+
+    def _validate_order_of_variables(self, df) -> [List[Exception]]:
+        """
+        Returns a list of exceptions that occured while doing checks
+        """
+        issues = []
+        cumulatives = df.query("measurement == 'cumulative'")
+        categories = list(cumulatives["category"].unique())
+        for category_small, category_large in self.ordered_validation_categories:
+            if category_small in categories and category_large in categories:
+                ok, problems = cross_section.cat1_ge_cat2(
+                    cumulatives,
+                    category_large,
+                    category_small,
+                    drop_levels=["unit", "category"],
+                )
+                if not ok:
+                    err = ValidateRelativeOrderOfCategoriesError(
+                        category_small=category_small,
+                        category_large=category_large,
+                        problems=problems,
+                    )
+                    issues.append(err)
+
+        return issues
+
     def validate(self, df, df_hist):
         """
         The `validate` method checks what the tentative clean data looks
         like and then checks whether the updates are sensible -- If they
         are then it returns True otherwise it returns False and stops
         the fetch->normalize->validate->put DAG
 
+        Raises Exceptions if validation fails
+
         Parameters
         ----------
         df : pd.DataFrame
             The data that is being proposed for put
         df_hist : pd.DataFrame
             Historical data
 
-        Returns
-        -------
-        validated : bool
-            Whether we have validated the data
         """
-        return True
+        issues = []
+        if utils.is_time_series(df):
+            issues.extend(self._validate_time_series(df))
+
+        issues.extend(self._validate_order_of_variables(df))
+
+        if len(issues) > 0:
```

Comment:
nit: would prefer `if issues:` [gsg explanation](https://google.github.io/styleguide/pyguide.html#2144-decision)

Reply:
I think I'm going to go against gsg on this one.


The first phrase

> Use the “implicit” false if possible,

grates at me relative to the zen of python

> Explicit is better than implicit.